vms/evm/subscriber.go

```diff
@@ -69,22 +117,150 @@ func NewWarpLogInfo(log types.Log) (*vmtypes.WarpLogInfo, error) {
 		SourceAddress:      log.Topics[3],
 		SourceTxID:         log.TxHash[:],
 		UnsignedMsgBytes:   log.Data,
+		BlockNumber:        log.BlockNumber,
 	}, nil
 }
 
 // forward logs from the concrete log channel to the interface channel
 func (s *subscriber) forwardLogs() {
 	for msgLog := range s.evmLog {
-		messageInfo, err := NewWarpLogInfo(msgLog)
+		messageInfo, err := s.NewWarpLogInfo(msgLog)
 		if err != nil {
-			s.logger.Info(
+			s.logger.Error(
 				"Invalid log. Continuing.",
 				zap.Error(err),
 			)
 			continue
 		}
-		s.log <- *messageInfo
+		s.logsChan <- *messageInfo
+	}
+}
+
+// Process logs from the given block height to the latest block
+// Cap the number of blocks requested from the client to MaxBlocksToProcess,
```

Comment:
May want to consider adding a TODO to remove the cap on the number of blocks processed here for now, given that we plan to follow up with that subsequent update.

Reply:
I think that cap will still be used in the future if we paginate calls to `eth_getLogs`. I'll leave this comment as is, and have captured that future work [here](https://github.com/ava-labs/awm-relayer/issues/29)